mountinfo/mountinfo_openbsd.go

```diff
@@ -0,0 +1,22 @@
+package mountinfo
+
+import "golang.org/x/sys/unix"
+
+func int8SliceToString(is []int8) string {
+	var bs []byte
+	for _, i := range is {
+		if i == 0 {
+			break
+		}
+		bs = append(bs, byte(i))
+	}
+	return string(bs)
+}
+
+func getMountinfo(entry *unix.Statfs_t) *Info {
+	var mountinfo Info
+	mountinfo.Mountpoint = int8SliceToString(entry.F_mntonname[:])
+	mountinfo.FSType = int8SliceToString(entry.F_fstypename[:])
+	mountinfo.Source = int8SliceToString(entry.F_mntfromname[:])
```

Comment:
I guess you forgot to switch to `unix.ByteSliceToString` here.

Reply:
No, this is correct.

In `syscall`, freebsd, darwin, and openbsd are all int8 arrays.
https://cs.opensource.google/go/go/+/refs/tags/go1.18.1:src/syscall/ztypes_freebsd_amd64.go;l=145
https://cs.opensource.google/go/go/+/refs/tags/go1.18.1:src/syscall/ztypes_darwin_amd64.go;l=100
https://cs.opensource.google/go/go/+/refs/tags/go1.18.1:src/syscall/ztypes_openbsd_amd64.go;l=117

But in `golang.org/x/sys/unix`, freebsd and darwin are byte arrays, only openbsd is int8 array.
https://cs.opensource.google/go/x/sys/+/33da011f:unix/ztypes_freebsd_amd64.go;l=129
https://cs.opensource.google/go/x/sys/+/33da011f:unix/ztypes_darwin_amd64.go;l=100
https://cs.opensource.google/go/x/sys/+/33da011f:unix/ztypes_openbsd_amd64.go;l=99

This is why I chose `syscall` before.

Reply:
Hmm, weird.

This is probably something that needs to be fixed in x/sys/unix. Something similar to e.g. https://go-review.googlesource.com/c/sys/+/359674 needs to be done.

If you have openbsd running, you can prepare a patch to x/sys/unix similar to the above; if not, I can try to do that or we can maybe kindly ask @tklauser

Reply:
Okay, I will make a patch to x/sys/unix for OpenBSD.

Reply:
Submitted.
https://go-review.googlesource.com/c/sys/+/401734